diff --git a/libfprint/meson.build b/libfprint/meson.build
index 0ca1767..11dc4e4 100644
--- a/libfprint/meson.build
+++ b/libfprint/meson.build
@@ -147,6 +147,10 @@ driver_sources = {
         [ 'drivers/synaptics/synaptics.c', 'drivers/synaptics/bmkt_message.c' ],
     'goodixmoc' :
         [ 'drivers/goodixmoc/goodix.c', 'drivers/goodixmoc/goodix_proto.c' ],
+    'goodix_gm168' :
+        [ 'drivers/goodix_gm168/goodix_gm168.c',
+          'drivers/goodix_gm168/goodix_proto.c',
+          'drivers/goodix_gm168/goodix_tls.c',
+          'drivers/goodix_gm168/gm168_log.c' ],
     'fpcmoc' :
         [ 'drivers/fpcmoc/fpc.c' ],
     'realtek' :
diff --git a/meson.build b/meson.build
index baafa19..f1964a9 100644
--- a/meson.build
+++ b/meson.build
@@ -140,6 +140,7 @@ default_drivers = [
     'upeksonly',
     'upekts',
     'goodixmoc',
+    'goodix_gm168',
     'nb1010',
     'fpcmoc',
     'realtek',
@@ -210,6 +211,7 @@ driver_helper_mapping = {
     'aes3500' : [ 'aeslib', 'aes3k' ],
     'aes4000' : [ 'aeslib', 'aes3k' ],
     'uru4000' : [ 'openssl' ],
+    'goodix_gm168' : [ 'openssl' ],
     'elanspi' : [ 'udev' ],
     'virtual_image'          : [ 'virtual' ],
     'virtual_device'         : [ 'virtual' ],
